main_test.go

```diff
@@ -230,6 +238,83 @@ var _ = Describe("Main", func() {
 			})
 		})
 	})
+
+	Describe("metricProcessor", func() {
+		type tokenJSON struct {
```

Comment:
no, it is not. You must use a new type to generate a json that will be parsed properly with a expire date. I will update the comment to explain that.